app/models/name/lifeform.rb

```diff
@@ -47,33 +47,91 @@ def validate_lifeform
 
   # Add lifeform (one word only) to all children.
   def propagate_add_lifeform(lifeform)
-    concat_str = Name.connection.quote("#{lifeform} ")
-    search_str = Name.connection.quote("% #{lifeform} %")
-    Name.connection.execute(%(
-      UPDATE names SET lifeform = CONCAT(lifeform, #{concat_str})
-      WHERE id IN (#{all_children.map(&:id).join(",")})
-        AND lifeform NOT LIKE #{search_str}
-    ))
-    Name.connection.execute(%(
-      UPDATE observations SET lifeform = CONCAT(lifeform, #{concat_str})
-      WHERE name_id IN (#{all_children.map(&:id).join(",")})
-        AND lifeform NOT LIKE #{search_str}
-    ))
+    concat_str = Name.connection.quote_string("#{lifeform} ")
+    search_str = Name.connection.quote_string("% #{lifeform} %")
+
+    for type in %w[name observation] do
+      update_manager = arel_update_add_lifeform(
+        type, concat_str, search_str
+      )
+      # puts(all_children.map(&:id).inspect)
+      # puts(update_manager.to_sql)
+      Name.connection.update(update_manager.to_sql)
+    end
   end
 
   # Remove lifeform (one word only) from all children.
   def propagate_remove_lifeform(lifeform)
-    replace_str = Name.connection.quote(" #{lifeform} ")
-    search_str  = Name.connection.quote("% #{lifeform} %")
-    Name.connection.execute(%(
-      UPDATE names SET lifeform = REPLACE(lifeform, #{replace_str}, " ")
-      WHERE id IN (#{all_children.map(&:id).join(",")})
-        AND lifeform LIKE #{search_str}
-    ))
-    Name.connection.execute(%(
-      UPDATE observations SET lifeform = REPLACE(lifeform, #{replace_str}, " ")
-      WHERE name_id IN (#{all_children.map(&:id).join(",")})
-        AND lifeform LIKE #{search_str}
-    ))
+    replace_str = Name.connection.quote_string(" #{lifeform} ")
```

Comment:
This looks equivalent to above, except there is no handy shorthand for REPLACE.  Still, it would seem worthwhile to use update_all.